Robot2019/src/main/java/frc/robot/lib/Limelight.java

```diff
@@ -0,0 +1,78 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.lib;
+
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class Limelight {
+  public enum Mode {
+    DIST, STEER, TARGET
+  }
+
+  private NetworkTableInstance inst;
+  private NetworkTable table;
+  /* http://docs.limelightvision.io/en/latest/networktables_api.html
+  tv = Whether the limelight has any valid targets (0 or 1)
+  tx = Horizontal Offset From Crosshair To Target (-27 degrees to 27 degrees)
+  ty = Vertical Offset From Crosshair To Target (-20.5 degrees to 20.5 degrees)
+  ta = Target Area (0% of image to 100% of image)
+  There are more values we could be using. Check the documentation.
+  */
+  private double tv, tx, ty, ta;
+
+  public Limelight() {
+    inst = NetworkTableInstance.getDefault();
+    table = inst.getTable("limelight");
+  }
+
+  // Adjusts the distance between a vision target and the robot. Uses basic PID with the ty value from the network table.
+  public double distanceAssist() {
+    tv = table.getEntry("tv").getDouble(0.0);
+    ty = table.getEntry("ty").getDouble(0.0);
+    SmartDashboard.putNumber("Crosshair Vertical Offset", ty);
+    ta = table.getEntry("ta").getDouble(0.0);
+    double adjustment = 0.0;
+    double area_threshold = 0.5;  // TODO: Set the desired area.
+    double Kp = 0.2;   // TODO: Set PID K value.
+
+    if (tv == 1.0) {
+      if (ta < area_threshold) {
+        adjustment += Kp * ty;
+      }
```

Comment:
This seems odd. It will move further faster if ty is larger, but ty will generally be larger if you are closer to the target. I also don't understand the point of area_threshold. As it stands, it will prevent moving backward if you are too close to the target. I recommend reviewing:
http://docs.limelightvision.io/en/latest/cs_estimating_distance.html (and doing the associated math)
and 
http://docs.limelightvision.io/en/latest/cs_autorange.html